File: examples/Example_16.java
package examples;

import java.io.*;
import java.util.*;
import com.pdfjet.*;
import com.pdfjet.fonts.*;

/**
 * Example_16.java
 */
public class Example_16 {
    public Example_16() throws Exception {
        PDF pdf = new PDF(
            new BufferedOutputStream(new FileOutputStream("Example_16.pdf")));
        pdf.setCompliance(Compliance.PDF_UA_1);
        pdf.setTitle("Text block with highlighted keywords");

        // Font f1 = new Font(pdf, SourceSerif4.Regular);
        // Font f1 = new Font(pdf, NotoSans.Regular);
        Font f1 = new Font(pdf, IBMPlexSans.Regular);
        f1.setSize(15f);

        Page page = new Page(pdf, Letter.PORTRAIT);

        Map<String, Integer> colors = new HashMap<String, Integer>();
        colors.put("Everyone", Color.red);
        colors.put("pay", Color.green);
        colors.put("freedom", Color.blue);

        // page.saveGraphicsState();

        GraphicsState gs = new GraphicsState();
        gs.setAlphaStroking(0.5f);                  // Stroking alpha
        gs.setAlphaNonStroking(0.5f);               // Non-Stroking alpha
        page.setGraphicsState(gs);

        String englishText = Content.ofTextFile("data/languages/english.txt");
        // f1.setSize(14f);
        TextBlock textBlock = new TextBlock(f1, englishText);
        textBlock.setLocation(100f, 50f);
        textBlock.setWidth(400f);
        // With a height the text that does not fit is cut; without one the
        // block is as tall as its text.
        textBlock.setHeight(450f);
        textBlock.setVerticalAlignment(Alignment.TOP);
        // textBlock.setVerticalAlignment(Alignment.BOTTOM);
        // textBlock.setVerticalAlignment(Alignment.CENTER);
        // textBlock.setTextAlignment(Alignment.CENTER);
        textBlock.setBackgroundColor(Color.whitesmoke);
        textBlock.setHighlightColors(colors);
        textBlock.setBorderColor(Color.black);
        float[] xy = textBlock.drawOn(page);

        page.setGraphicsState(new GraphicsState()); // Reset GS
        // page.restoreGraphicsState();

        Rect rect = new Rect(xy[0], xy[1], 20f, 20f);
        rect.setBorderColor(Color.black);
        rect.drawOn(page);

        pdf.complete();
    }

    public static void main(String[] args) throws Exception {
        long time0 = System.currentTimeMillis();
        new Example_16();
        long time1 = System.currentTimeMillis();
        System.out.printf("Example_16 => %4d ms%n", time1 - time0);
    }
}   // End of Example_16.java

File: examples/Example_16.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Diagnostics;
using PDFjet.NET;

/**
 *  Example_16.cs
 *
 */
public class Example_16 {
    public Example_16() {
        PDF pdf = new PDF(new BufferedStream(
                new FileStream("Example_16.pdf", FileMode.Create)));

        // Font f1 = new Font(pdf, CoreFont.HELVETICA);
        // Font f1 = new Font(pdf, "fonts/SourceSansPro/SourceSansPro-Regular.otf");
        // Font f1 = new Font(pdf, "fonts/SourceCodePro/SourceCodePro-Regular.ttf");
        // Font f1 = new Font(pdf, "fonts/OpenSans/OpenSans-Regular.ttf");
        Font f1 = new Font(pdf, "fonts/RedHatText/RedHatText-Regular.ttf");
        f1.SetSize(14f);

        Page page = new Page(pdf, Letter.PORTRAIT);

        Dictionary<String, Int32> colors = new Dictionary<String, Int32>();
        colors["Lorem"] = Color.blue;
        colors["ipsum"] = Color.red;
        colors["dolor"] = Color.green;
        colors["ullamcorper"] = Color.gray;

        GraphicsState gs = new GraphicsState();
        gs.SetAlphaStroking(0.5f);      // Set alpha for stroking operations
        gs.SetAlphaNonStroking(0.5f);   // Set alpha for nonstroking operations
        page.SetGraphicsState(gs);
/*
        f1.SetSize(72f);
        TextLine text = new TextLine(f1, "Hello, World");
        text.SetLocation(50f, 300f);
        text.DrawOn(page);
*/
        String latinText = File.ReadAllText("data/latin.txt");

        f1.SetSize(14f);
        TextBox textBox = new TextBox(f1, latinText);
        textBox.SetLocation(100f, 50f);
        textBox.SetWidth(400f);
        // If no height is specified the height will be calculated based on the text.
        textBox.SetHeight(450f);
        textBox.SetWidth(400f);
        textBox.SetHeight(450f);
        textBox.SetTextDirection(Direction.LEFT_TO_RIGHT);
        // textBox.SetTextDirection(Direction.BOTTOM_TO_TOP);
        // textBox.SetTextDirection(Direction.TOP_TO_BOTTOM);
        // textBox.setVerticalAlignment(Align.TOP);
        textBox.SetVerticalAlignment(Align.BOTTOM);
        // textBox.SetVerticalAlignment(Align.CENTER);
        // If no height is specified the height will be calculated based on the text.
        textBox.SetBgColor(Color.whitesmoke);
        textBox.SetTextColors(colors);
        textBox.SetBorders(true);
        textBox.SetBorder(Border.ALL);
        float[] xy = textBox.DrawOn(page);

        page.SetGraphicsState(new GraphicsState()); // Reset GS

        Box box = new Box();
        box.SetLocation(xy[0], xy[1]);
        box.SetSize(20f, 20f);
        box.DrawOn(page);

        pdf.Complete();
    }

    public static void Main(String[] args) {
        Stopwatch sw = Stopwatch.StartNew();
        long time0 = sw.ElapsedMilliseconds;
        new Example_16();
        long time1 = sw.ElapsedMilliseconds;
        sw.Stop();
        TextUtils.PrintDuration("Example_16", time0, time1);
    }
}   // End of Example_16.cs
